ui/src/sqlDialog.tsx

```diff
@@ -0,0 +1,88 @@
+import DownloadIcon from "@mui/icons-material/Download";
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import IconButton from "@mui/material/IconButton";
+import { EntityInstancesApiContext } from "apiContext";
+import React, { useContext, useEffect, useState } from "react";
+import { Dataset } from "./dataset";
+
+type SqlDialogProps = {
+  dataset: Dataset;
+};
+
+export default function SqlDialog(props: SqlDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [sql, setSql] = React.useState<string | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const api = useContext(EntityInstancesApiContext);
+
+  useEffect(() => {
```

Comment:
nit: should this use the async / fetch pattern (I believe you added this elsewhere), e.g. for XHR cleanup on navigate?